src/rust/engine/src/nodes.rs

```diff
@@ -175,17 +176,21 @@ impl WrappedNode for Select {
                 .collect::<Vec<_>>(),
             )
             .await?;
-            context
+            match context
```

Comment:
nit: I think it's generally cleaner to assign complex expressions to a variable and then match on the variable, rather than having a complex expression  as the target of a match.

Reply:
Done!